presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,19 +721,24 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int[] leftCodePoints = castToCodePoints(left);
-        int[] rightCodePoints = castToCodePoints(right);
-
         checkCondition(
-                leftCodePoints.length == rightCodePoints.length,
+                safeCountCodePoints(left) == safeCountCodePoints(right),
                 INVALID_FUNCTION_ARGUMENT,
-                "The input strings for Hamming distance must have the same length");
+                "The input strings to hamming_distance function must have the same length");
 
         int distance = 0;
-        for (int i = 0; i < leftCodePoints.length; i++) {
-            if (leftCodePoints[i] != rightCodePoints[i]) {
+        int leftPosition = 0;
+        int rightPosition = 0;
+        while (leftPosition < left.length()) {
```

Comment:
Presto generally choose to be strict in terms of behaviors under exceptional cases. However, in the case of string functions, we realized that failing on invalid UTF8 will basically result in query failure for any non-trivial queries due to inevitable bad data. Therefore, we have chosen to be lenient (garbage in garbage out) in such cases.